@shared/hooks/useFirstTokenID.ts

```diff
@@ -0,0 +1,30 @@
+import { zdk } from '@shared'
+import { useMemo } from 'react'
+import useSWR from 'swr'
+import { NetworkInput } from 'utils/network'
+
+export function useFirstTokenID(collectionAddress: string) {
+  const { data: tokenData, error: firstTokenDataError } = useSWR(
+    [`firstTokenFor-${collectionAddress}`, collectionAddress],
+    (_, collectionAddress) =>
+      zdk.token({
+        token: {
+          address: collectionAddress,
+          tokenId: '0',
```

Comment:
what would be response if collection begins with tokenId=1?

Reply:
@olegakbarov 

```
{token: null}
```

So this is handled by the check on tokenData when setting firstTokenID:

```
tokenData?.
```

Reply:
(Have confirmed this on some contracts locally, btw
)